[KarKuen] iP

KarKuen opened the pull request:
#IP
Friday ChatBot that stores checklists

tasks available to be stored within the checklist:
- todo
- event
- deadline

todo task requirements:
1. todo keyword
2. description

event task requirements:
1. event keyword
2. description
3. from & to keywords

deadline task requirements:
1. deadline keyword
2. description
3. by keyword

the location of the stored checklist file is under the user's **desktop**
```java
private static String home = System.getProperty("user.home");
    private static String filePath = String.valueOf(java.nio.file.Paths.get(home, "Desktop", "taskList"));
```
> [!NOTE]
> user.home is required to retrieve user's _unique desktop id_


- [x] todo
- [x] event
- [x] deadline

this description follows the GitHub Flavored Markdown [GFMD](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) :+1:

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

KarKuen committed:
Renamed Duke to Friday.
Implemented a simple greeting.

KarKuen committed:
Implemented a feature for Friday to echo the inputs typed by the user.

KarKuen committed:
Friday now stores the text inputs and display them back to the user when requested.

KarKuen committed:
Implemented a Task Class.
Implemented a checkbox feature.

KarKuen committed:
Added Todo, Deadline, and Event classes.
Included a function to abstract the printing of horizontal lines.

KarKuen committed:
Added TextUITesting

KarKuen committed:
Added 2 Exception Handling.

KarKuen committed:
Implemented two Exception Handling.

KarKuen committed:
Implemented a delete function.

KarKuen committed:
Add tasks and dukeexceptions packages, and file saving in TaskList file.

Moved classes Task, TodoTask, EventTask, DeadlineTask, DukeException into packages.

Created a TaskList file in the User's Desktop that stores allTasks before exiting the program.

KarKuen committed:
Add tasks and dukeexceptions packages, and file saving in TaskList file

Moved classes Task, TodoTask, EventTask, DeadlineTask, DukeException into packages

Created a TaskList file in the User's Desktop that stores allTasks before exiting the program

KarKuen committed:
Add LocalDateTime to verify the format of the DeadlineTask date

KarKuen committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Friday.java
#	src/main/java/dukeexceptions/DukeException.java
#	src/main/java/tasks/DeadlineTask.java

KarKuen committed:
Update master branch to be consistent with branch-Level-8

KarKuen committed:
Refactor the code to extract out Ui, Storage, Parser, TaskList class

KarKuen committed:
Extract Command, AddCommand, DeleteCommand, ExitCommand, BasicCommand class

KarKuen committed:
Extract Command, AddCommand, DeleteCommand, ExitCommand, BasicCommand classes

KarKuen committed:
Organise all classes into packages

KarKuen committed:
Merge branch 'add-gradle-support'

KarKuen committed:
Add JUnit tests

KarKuen committed:
Package the App as a JAR file

KarKuen committed:
Edit Ui readCommand function to not create a new Scanner Object every time it is called

KarKuen committed:
Add JavaDoc comments to the code

KarKuen committed:
Tweak the code to comply with a coding standard

KarKuen committed:
Add find action to search for a keyword in the task description

KarKuen committed:
Merge branch 'branch-A-CodingStandard'

KarKuen committed:
Merge branch 'branch-Level-9'

KarKuen committed:
Fix runtest.sh file location

KarKuen committed:
Add SuppressWarnings for generic ArrayList allTasks

ambertan77 commented on src/main/java/friday/Friday.java:
```diff
@@ -0,0 +1,55 @@
+        }
+    }
+
+    public void run() {
```
Maybe can specify what is being ran in the method name here?

ambertan77 commented on src/main/java/friday/command/AddCommand.java:
```diff
@@ -0,0 +1,41 @@
+
+import java.time.LocalDateTime;
+
+public class AddCommand extends Command {
```
`AddCommand` (and `DeleteCommand`) can be interpreted as a verb, maybe there would be better ways to phrase it?

ambertan77 commented on src/main/java/friday/parser/Parser.java:
```diff
@@ -0,0 +1,49 @@
+package friday.parser;
+
+import friday.command.*;
```
I think it might be better to specify the import classes explicitly here.

ambertan77 commented on src/main/java/friday/dukeexceptions/DukeException.java:
```diff
@@ -0,0 +1,18 @@
+package friday.dukeexceptions;
+
+public class DukeException extends Exception {
```
Can I ask why is the name of the exception `DukeException`? I thought it would be `FridayException` since Friday is the name of your chatbot.

KarKuen replied:
You're right! I should change the exception names, thank you!

ambertan77 commented on src/main/java/friday/parser/Parser.java:
```diff
@@ -0,0 +1,49 @@
+            }
+        }
+        String action = fullCommand.split(" ")[0];
+        if ((action.compareTo("todo") == 0)
```
Any reason why you decided to use `compareTo` instead of `equals` here?

KarKuen replied:
There wasn't any reason why I picked compareTo() over equals(), but I'm trying to stay consistent and only use either of them.

ambertan77 commented on src/main/java/friday/ui/Ui.java:
```diff
@@ -0,0 +1,69 @@
+     * Returns a farewell message.
+     * @return The farewell message.
+     */
+    public static void farewell() {
```
I think it might be better to name this method using a verb, e.g. `bidFarewell()`.

KarKuen replied:
This is a good suggestion. Thanks!

ambertan77 reviewed (commented):
Overall, your code is clean and easy to understand. Nice job :)

KarKuen committed:
Update dukeexceptions to fridayexceptions

potatodudedude commented on src/main/java/friday/Friday.java:
```diff
@@ -0,0 +1,55 @@
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
```
According to [Java coding standard](https://se-education.org/guides/conventions/java/intermediate.html), you need a Javadoc for every public class and public method (with some exceptions). 😢 You can follow the [checkstyle guide](https://se-education.org/guides/tutorials/checkstyle.html), so it'll automatically remind you to put a Javadoc.

potatodudedude commented on src/main/java/friday/fridayexceptions/FridayException.java:
```diff
@@ -0,0 +1,18 @@
+        this.message = message;
+    }
+
+    /**
+     * Returns the message associated with the DukeException.
+     * @return The message to be returned.
+     */
```
No need to put JavaDoc in front of simple getter/setter methods. Save some finger power for other stuff!

potatodudedude commented on src/main/java/friday/command/ExitCommand.java:
```diff
@@ -0,0 +1,28 @@
+            //save allTasks into TaskList file
+            Storage.saveFile(TaskList.returnList());
+            ui.farewell();
+        } catch (IOException e){
```
```suggestion
        } catch (IOException e) {
```
Minor coding standard violation, as mentioned earlier you can set up a checkstyle so you don't need to comb through everything yourself.

potatodudedude commented on src/main/java/friday/storage/Storage.java:
```diff
@@ -0,0 +1,58 @@
+        }
+    }
+
+    /**
+     * Saves all tasks into the TaskList file.
+     * @param allTasks The tasks to be saved into the TaskList file.
+     * @throws IOException When the tasks cannot be written into the TaskList file.
+     */
```
```suggestion
    /**
     * Saves all tasks into the TaskList file.
     *
     * @param allTasks The tasks to be saved into the TaskList file.
     * @throws IOException When the tasks cannot be written into the TaskList file.
     */
```
Need a line between description and parameters. I think all your Javadocs have this error.  Otherwise this is a pretty good Javadoc. 👍

potatodudedude reviewed (commented):
Biggest thing to do in terms of coding standard are your JavaDocs. Otherwise, it's just a few minor violations here and there. 👍 All the best!

KarKuen committed:
Add a GUI to the chatbot

KarKuen committed:
Add gui package

KarKuen committed:
Generate new JAR file

KarKuen committed:
Remove redundant methods and packages

KarKuen committed:
Use assert feature to document assumptions about existing taskList file's string format

The taskList file is assumed to be unedited by the user.
Any changes to the string formatting will result in an error loading in the Tasks.

Assert checks the string formatting for each task.

Let's,
* check the string format of TodoTask to follow
	"[T][ ] description"
* check the string format of TodoTask to follow
	"[E][ ] description  (from: startDate to: endDate)"
* check the string format of TodoTask to follow
	"[D][ ] description  (by: month date year time)"

KarKuen committed:
Remove redundant methods and packages

KarKuen committed:
Improve code quality

There are several magic literals set to distinguish different tasks.
Return statements of string concatenation is not obvious.

Let's,
* implement SLAP to have the same level of abstraction
* avoid magic numbers with static final variables
* make the happy path prominent

KarKuen committed:
Merge pull request #1 from KarKuen/branch-A-Assertions

Branch a assertions

KarKuen committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/friday/tasklist/TaskList.java
#	src/main/java/friday/tasks/EventTask.java

KarKuen committed:
Merge pull request #2 from KarKuen/branch-A-CodeQuality

Branch a code quality

KarKuen committed:
Add C-Priority extension

Priority tagging allows to filter TaskList effectively.

Add prioritise function to set the task's priority field and add
plist function to filter all tasks in accordance with their priority.

Using an ArrayList<String> to store all tasks locally, priority cannot
be stored after program executes without altering the string value of
the tasks to include its priority levels.

KarKuen committed:
Update code to follow coding standard

KarKuen committed:
Remove mentions of duke

KarKuen committed:
Improve code quality

KarKuen committed:
Add createToDo(), createDeadLine(), createEvent() methods

KarKuen committed:
Update JUnit tests

KarKuen committed:
Improve code quality

KarKuen committed:
Fix unexpected behaviour

KarKuen committed:
Add GUI title and Ui.png

KarKuen committed:
Improve coding standard

KarKuen committed:
Add README.md

KarKuen committed:
Fix unexpected behaviours

KarKuen committed:
Change to relative filepath

KarKuen committed:
Add user guide

KarKuen committed:
Add link to JAR file

KarKuen committed:
Add tests
